include/ametsuchi/io/cvfs.h

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <ametsuchi/io/vfs.h>
+#include <ametsuchi/pager/page.h>
+#include <ametsuchi/cache.h>
+#include <cstdio>
+
+namespace ametsuchi {
+namespace io {
+
+enum { CVFS_MAX_FILE_SIZE = 4096 * 16, CVFS_MAX_CACHE_SIZE = 256, };
+
+typedef FILE* FileHandler;
+
+/*
+* This class perform the real basic implementation of VFS
+* All operations are performed via C-based i/o function
+* todo?: respect max_file_size & WAL
+*/
+class CVFS : public virtual VFS {
```

Comment:
1. The implementation keeps everything in one single file, so it might be called `FileVFS` or at least `StdioVFS`.
2. An instance of this class represents a _single_ file, not the whole file system; therefore, `Store` or `File` might be more descriptive.